File: create-round-icon.js
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

async function createRoundIcon(inputPath, outputPath, size) {
  try {
    // 画像のメタデータを取得
    const metadata = await sharp(inputPath).metadata();
    const { width, height } = metadata;
    
    // 正方形にクロップ（中央部分を取得）
    let left = 0;
    let top = 0;
    let cropSize;
    
    if (width > height) {
      // 横長の場合、中央を正方形に
      cropSize = height;
      left = Math.floor((width - height) / 2);
    } else {
      // 縦長の場合、中央を正方形に
      cropSize = width;
      top = Math.floor((height - width) / 2);
    }
    
    // 正方形にクロップしてリサイズ
    let image = await sharp(inputPath)
      .extract({ left, top, width: cropSize, height: cropSize })
      .resize(size, size, { fit: 'cover' });
    
    // 丸いマスクを作成
    const svgMask = `
      <svg width="${size}" height="${size}">
        <circle cx="${size / 2}" cy="${size / 2}" r="${size / 2}" fill="white"/>
      </svg>
    `;
    
    // 丸いマスクを適用
    const rounded = await image
      .composite([
        {
          input: Buffer.from(svgMask),
          blend: 'dest-in'
        }
      ])
      .png()
      .toBuffer();
    
    // 保存
    await fs.promises.writeFile(outputPath, rounded);
    console.log(`✓ 丸いアイコンを作成しました: ${outputPath} (${size}x${size})`);
    
    return true;
  } catch (error) {
    console.error(`エラーが発生しました (${outputPath}):`, error.message);
    return false;
  }
}

async function main() {
  const inputPath = 'app/icon.png';
  
  // 複数のサイズで作成
  const sizes = [
    { size: 512, path: 'app/icon.png' },
    { size: 512, path: 'public/favicon.png' },
    { size: 180, path: 'public/apple-touch-icon.png' },
    { size: 32, path: 'public/favicon-32x32.png' },
    { size: 16, path: 'public/favicon-16x16.png' },
  ];
  
  console.log('丸いアイコンの作成を開始します...\n');
  
  for (const { size, path: outputPath } of sizes) {
    await createRoundIcon(inputPath, outputPath, size);
  }
  
  console.log('\n完了しました！');
}

main().catch(console.error);

File: create-round-icon.py
#!/usr/bin/env python3
"""
画像を丸くトリミングしてファビコンを作成するスクリプト
"""
from PIL import Image, ImageDraw
import sys
import os

def create_round_icon(input_path, output_path, size=None):
    """画像を丸くトリミングして保存"""
    try:
        # 画像を開く
        img = Image.open(input_path).convert("RGBA")
        
        # 元の画像サイズを取得
        width, height = img.size
        
        # 正方形にクロップ（中央部分を取得）
        if width > height:
            # 横長の場合、中央を正方形に
            left = (width - height) // 2
            top = 0
            right = left + height
            bottom = height
        else:
            # 縦長の場合、中央を正方形に
            left = 0
            top = (height - width) // 2
            right = width
            bottom = top + width
        
        # 正方形にクロップ
        img_cropped = img.crop((left, top, right, bottom))
        
        # サイズ指定がある場合はリサイズ
        if size:
            img_cropped = img_cropped.resize((size, size), Image.Resampling.LANCZOS)
        else:
            size = img_cropped.size[0]
        
        # 丸いマスクを作成
        mask = Image.new('L', (size, size), 0)
        draw = ImageDraw.Draw(mask)
        draw.ellipse((0, 0, size, size), fill=255)
        
        # 丸いマスクを適用
        output = Image.new('RGBA', (size, size), (0, 0, 0, 0))
        output.paste(img_cropped, (0, 0))
        output.putalpha(mask)
        
        # 保存
        output.save(output_path, 'PNG')
        print(f"丸いアイコンを作成しました: {output_path} ({size}x{size})")
        return True
        
    except ImportError:
        print("PIL (Pillow) がインストールされていません。")
        print("インストール方法: pip3 install Pillow")
        return False
    except Exception as e:
        print(f"エラーが発生しました: {e}")
        return False

if __name__ == "__main__":
    input_path = "app/icon.png"
    
    # 複数のサイズで作成
    sizes = [
        (512, "app/icon.png"),  # Next.js用
        (512, "public/favicon.png"),  # 公開用
        (180, "public/apple-touch-icon.png"),  # Apple Touch Icon用
        (32, "public/favicon-32x32.png"),  # 標準ファビコン用
        (16, "public/favicon-16x16.png"),  # 小さいファビコン用
    ]
    
    for size, output_path in sizes:
        create_round_icon(input_path, output_path, size)
